bokehjs/src/coffee/models/ranges/factor_range.coffee

```diff
@@ -70,6 +70,13 @@ export map_three_levels = (factors, outer_pad, inner_pad, factor_pad, offset=0)
 export class FactorRange extends Range
   type: 'FactorRange'
 
+  `
+  factors: string[]
+
+  mids: string[] | undefined
+  tops: string[] | undefined
+  `
```

Comment:
@mattpap What going on here with the backticks?

Reply:
Those backticks are CoffeeScript's syntax for injecting raw JavaScript. I hijacked this to inject TypeScript, to fill-in missing type information. Otherwise modules being rewritten would require many more instances of `any` type sprinkled around.